### ISSUE TITLE
form. A bug with selecting text with shift +  right/left arrow keys

### ISSUE THREAD

**Issue by peter1384** [samtupy/nvgt#63]
so, i'll provide steps to reproduce to prevent confusion, and so you can understand the problem properly
1: type "hello" in the editbox
2: As this is not reproduceable with the first character, press right arrow once or twice to step on the leter "E" or "L" of the hello, We are focusing on "E" in this example.
3: press shift + right arrow once, you'll hear "e  Selected".
4: press shift +left arrow.
In nvgt form: You'll hear "h  Selected", Which is not correct and shouldn't be like this.
In windows notepad: You'll hear "e unselected", Which is correct and what should happen.
This also happens with selecting a leter with shift +left arrow and then you try to unselect it with shift + right arrow, That's also broken same as left arrow which steps provided above. So, please fix these text selection bugs. Thanks!

**Comment by peter1384** [comment]
hi @samtupy While my first issue related to change the search text dialog and make it like notepad is not very important And I just openned it because it's nice to have it like that generally but it can be done at any point later, Can you please investigate my 2 last issues which one is this, And the things requested in issue of modifying go to dialog? I need these, And moste importantly, I don't whish anyone else except you to fix these form issues, Because you are the gratest form modifyer as i've seen in the programming history ever, as an example, I love nvgt's form's words and characters ehko how if you wright a word and press space, as many times as you want to read your word again, You can delete the space and press space again to hear your word again, where as in nvda you can't, your word is being red to you only the first time you press space, then the buffer will be cleared, which means pressing backspace and space again won't read your word again, you have to read your word in another way. This was a tiny example of how your form is nice, This is not only stikked to nvgt, I used to see stw's forms even back to the time stw was stil in bgt, And i'd have a deap breath in my chest and be like, oh god, how nice, but sadly in that time asking you for any peace of code would bring a permanent ban LOL LOL, so, I don't want any other contribution for these form related, except you yourself. Thanks!

**Comment by peter1384** [comment]
+ I can't forget, can't forget that how much time I took messing around with bgt's form to get something in to work, And everytime I faled, except only once I managed to make it so that when focused on the last line, pressing down arrow would speeck the line and not be silent, And that was again broken because when focused on the last character the down arrow would say blank, focus didn't move properly so if we wanted to add any other text at the end of the line we needed to move the focus by pressing right arrow until end of the line, oh god!

**Comment by peter1384** [comment]
@samtupy another note: This bug only happens with leter selection which is done with shift+left/right arrow keys, The word selection which is done with ctrl+shift+left/right arrow keys doesn't have any bug, and works fine as expected

**Comment by peter1384** [comment]
Now, I'll report the word selection bugs which are fue of them, here in this comment
first bug
1: type "hello" in the editbox, with out adding any space or anything at the end of it
2: press left arrow until mooved to the first leter of the hello which is h
3: press ctrl+shift+right arrow, you'll hear "hello Selected"
4: press ctrl+shift+left arrow, you'll hear "hell Unselected", Which means the last character is not spoke
But you should've heard "hello Unselected"
bug 2
1: type "hello" in the editbox, with out adding any space or anything at the end of it, note that when you add any space or any of the audio form word seperators at the end of it, This bug will disappear, So just type hello
2: press ctrl+right/left arrow keys to read the word, you'll just hear blank, But if any of the audio form word seperators are at the end of the hello, word movement will work fine and will read your hello to you because then it asumes that there are more than only 1 word, Currently it has this bug when it asumes there is only 1 word

**Comment by peter1384** [comment]
I hope We investigate and fix these bugs soon because they are so complicated and advance and stuf

**Comment by braillescreen** [comment]
Hi there.

It would be much appreciated if you edit issue comments directly rather than post multiple comments at the same time; while also keeping them pretty directly related to the issue at hand. GitHub discussions are available for long-winded responses, but I believe the engine's core contributors would much appreciate being able to jump between various problems they can fix and immediately have information they need to fix it.

Additionally, please keep in mind that each new comment you send, it sends an email each time! This makes it a bigger struggle going through emails, in general.

Thanks much!

**Comment by peter1384** [comment]
@braillescreen hi, i'm sorry if sometimes I posted multipul comments, But yeah, I also edit my comments when I remember something new to add in my comment.

**Comment by amirmahdifard** [comment]
@samtupy hi. This issue is long standing, And I beleave you can fix these because you are really good at virtual forms. Can you please take a time to fix these, We want to use. Thank you so much!

**Comment by amirmahdifard** [comment]
i'm being annoying and sending too many messages, but i'm really sad becausebI really like nvgt's form and if these selection bugs gets fix, It is much better than notepad and any real windows editboxes for me

**Comment by Jonathans859** [comment]
> i'm being annoying and sending too many messages, but i'm really sad becausebI really like nvgt's form and if these selection bugs gets fix, It is much better than notepad and any real windows editboxes for me


Bro, seriously? You say yourself you're sending too many messages and are annoying, so what do you think will happen? This is a free time project; no one gets money for doing this. No one has to fix your issue within a certain time frame. And think there are tons of other issues which need to be addressed instead of a bug in an old, buggy BGT include which will be recoded eventually. Also if some virtual edit box is better than NotePad you should have other problems, and I know this might come over as harsh but bruh, I'm also not repeating the same message every half week. It won't make your issue more important, it just clutters notifications. So thanks for your patience.

**Comment by braillescreen** [comment]
> i'm being annoying and sending too many messages, but i'm really sad becausebI really like nvgt's form and if these selection bugs gets fix, It is much better than notepad and any real windows editboxes for me

Usually, a good time to wait before bumping the issue again is around 3 months, I'd think. First, though, I'd attempt to use other means (such as the [NVGT Forum](https://forum.nvgt.gg) or the Discord server.

Bumping issues like this isn't always bad; reminders are sometimes a wonderful thing.

**Comment by amirmahdifard** [comment]
@braillescreen hi. Yeah I truely understand this and just commented because I saw this yesterday after a long time again, because I remembered that @samtupy said he will fix these in the comming days, And I was very excited because sam already recoded the bgt's buggy forms, which is now this, even better than sbyw's editboxes that they made their own virtual editboxes from scrach to avoid bgt's buggy form, and they are the only ones who have done in bgt game developers history, And now this form is even better than that! After the creation of this issue, I saw a fue times sam modifyed some things in the form. And now, It would be really really much thankful if he fixes these. I even know where have to be modifyed to fix these, a fue modifications to void hilight left and hi light right would do it, but I my self was never good at forms stuf at all, because In those times nvgt wasn't a thing yet, I tryed to fix bgt's messy form And I left it aftr long trys. But since @samtupy recoded it already so faar and it's now nice as such, he can fix these at fingertips, so this is being better of anything. But yesterday I got a bit of sadness about why this issue is stil not fixed about a form which everyone enjoys by just looking at it! these are text selection bugs and that's what is being preventing me from make it my mane text editor, because cannot select and deselect leters/words as expected, and that makes me sad. Hopefully, this gets some attention. Thanks!

**Comment by amirmahdifard** [comment]
@samtupy hi. I tryed to show these bugs even to chat gpt, and that cannot fix that even. the thing that I know is, the void move word right, checks for the seperaters to seperate the words and move word by word when using controle+right arrow. But, it's somehow desighned that skips the first word and starts moving from the second word, so when having a single word with out any seperater and trying to move right, it says blank because the first word is completely skipped. And this bug is only with word movement, not word selection.

**Comment by amirmahdifard** [comment]
@samtupy hello, I want to ask you something else, hopefully you'll answer this time.
You said that the form will be remade again from scrach, does that mean that all current features such as go to line, find text feachers will be lost or all of them will be restored? also, why you have to do extra work? You've already fixed all of the form, just there is only 1 or too small bug related to leter selection, otherwise you've already fixed the bugs of bgt's audio form all, so what's the exact point of reright of form exactly? I really don't know and this has been a question for me from that day you added the form reright plan to the todo documentation of the nvgt documentations

**Comment by samtupy** [comment]
NVGT's form will be rewritten to provide for an object oriented structure where handles to control objects are used, not control inndexes. This way controls can be added, removed, rearranged and modified without needing to keep track of control indexes that would change in that case. There are other issues with the form, such as it's lack of support for translation, it's lackluster event handling system and it's generally legacy coding practices. Any new form we recode would act similarly to the old one, but would have a completely different API and would be much cleaner and more maintainable. Yes things like the go to line and find dialogs would function as before. The old form would still exist for legacy code, but would not be as maintained. I imagine we'll copy quite a bit of logic from the old form to the new one so as to preserve carefully constructed bugfixes and text selection logic over the years, but as it stands the current form is getting ever messier to maintain as we keep adding on to this monolith.

**Comment by amirmahdifard** [comment]
@samtupy hi. Thanks for your final answer. I hope anything that will happen will appear as good hopefully, But also I want to bring something in to your attention.
it's not the actual and correct topic for this i'm bringing up, But the only thing I understand so faar is that you're not reachable easily. Even tho you have your contact information on your website, it looks like anything is not being reached to you, when we message you throo email, skipe or anything, So i'm just saying it hear. Really I shouldn't or may be I can just leave it, but sadly I cannot. I mean my emotional and fealings doesn't allow it. Please, Poste something, an update, a new final blog about stw that what you're gonna do with it at the final last. You made a game that everyone seriously loved it, You actually managed to bring everyone to love your game, from people who would play constant batle, to people who plays sbyw, to that dl, etc. your game was the single game that everyone seriously loved it, and now you took it down, even with your very very long blog poste, the only thing anyone didn't understood from your poste, is that what are you gonna do with that game sam. See, due to the faqt that you said may be you will bring it back, you didn't allow stw to be run on any server, so seriously you made hearing stw's sound once more anyone's dream. The other day I was hearing someone saying my big wish is that I want to know will I ever play stw again once more in my life before my death, witch was looked very stupid that someone is being bound to an audio game that much, and everyone is saying anything about it. I'm sure you haven't had a look at your blog for a long time, to see what people are saying. I'm really sorry sam, I don't wish to be noizy, but i'll do my best to reach this message in to your hand, so you can read it. sam. Please, poste one more blog about stw, wether you want to open source it so that everyone can play it, but you know that stw is known enough so that anyone cannot say anything about it like, stw is being built and developed by someone else except sam, etc. So, wether you want to open source it and let players play it once more in their lives as they said, or you want to bring it back, or just destroying everyone's hope by saying stw is dyed forever, so people can stop, someone says I know sam will bring it back, another says I don't think so. So please, poste another blog for stw letting everyone know what are you gonna do finally. You've made a game that everyone loves it, so your responceability for it haven't gon away by shutting it down, you are stil responceable because so many people are every day, looking at your website looking for another news of stw and they see there's nothing. So please do something about this. If this message is not reached to you, i'll again try to reach it to you and won't step back, until i'm sure at least you've red this message completely, even with the faqt that you won't take any action, just the matter of you've read this message that seriously every day morning, when waking up, before bathroom, brekfast, checking emails and hour own dayly work, we check samtupy.com blogs section and see there's nothing. it's not me, it's seriously more than what you can think. Thanks. let me know you've got the point.

**Comment by amirmahdifard** [comment]
@samtupy hi. Some of my persian friends asked me something because they didn't know how to contact you them selves. if you saw this message please answer.
apparently they've talked to you before and you gave them the right to run your redspot game for 3 months or something like that they told me. And now, They want to ask you are they allowed to run your redspot game again and develop it more and play it? they told me to tell you that they won't change the name of the game and won't remove your name from it or any of this, they just want to run it and play it, but they beleave on god and they cannot do that with out your rights. So, please let me know your answer so I can tell them. Thanks.

**Comment by aryanchoudharypro** [comment]
If you want to contact him you can do it in a few places but please don't fill the GitHub issues with stuff like this. It doesn't belong here. This GitHub is for NVGT discussions only.

**Comment by ethindp** [comment]
@amirmahdifard I have hidden your last two comments as they are irrelevant to this particular issue. In future, please do not spam issues with off-topic content. If you continue to do so, I (at minimum) will start locking those issues. This is not the first warning you have received from other contributors of the project and users over this inappropriate behavior. I would strongly encourage you to heed them; I will not warn you again.
